runtime/interp.c

```diff
@@ -290,27 +322,28 @@ value caml_interprete(code_t prog, asize_t prog_size)
 #ifdef DEBUG
  next_instr:
   if (caml_icount-- == 0) caml_stop_here ();
-  CAMLassert(sp >= Caml_state->stack_low);
-  CAMLassert(sp <= Caml_state->stack_high);
+  CAMLassert(Stack_base(domain_state->current_stack) <= sp &&
+         sp <= Stack_high(domain_state->current_stack));
 #endif
   goto *(void *)(jumptbl_base + *pc++); /* Jump to the first instruction */
 #else
   while(1) {
 #ifdef DEBUG
     caml_bcodcount++;
     if (caml_icount-- == 0) caml_stop_here ();
-    if (caml_trace_level>1) printf("\n##%" ARCH_INTNAT_PRINTF_FORMAT "d\n",
-                                   caml_bcodcount);
-    if (caml_trace_level>0) caml_disasm_instr(pc);
-    if (caml_trace_level>1) {
+    if (caml_params->trace_level>1)
+      printf("\n##%" ARCH_INTNAT_PRINTF_FORMAT "d\n", caml_bcodcount);
+    if (caml_params->trace_level>0) caml_disasm_instr(pc);
+    if (caml_params->trace_level>1) {
       printf("env=");
       caml_trace_value_file(env,prog,prog_size,stdout);
       putchar('\n');
       caml_trace_accu_sp_file(accu,sp,prog,prog_size,stdout);
       fflush(stdout);
     };
-    CAMLassert(sp >= Caml_state->stack_low);
-    CAMLassert(sp <= Caml_state->stack_high);
+    CAMLassert(Stack_base(domain_state->current_stack) <= sp &&
+           sp <= Stack_high(domain_state->current_stack));
```

Comment:
Move it into its own `CAMLassert` as in trunk.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/eebbb5ab5a3be66817892b032949ca9387e4df4f